crates/starfish/core/src/linearizer.rs

```diff
@@ -679,44 +582,48 @@ mod tests {
         }
     }
 
-    /// This test will run the linearizer with GC disabled (gc_depth = 0) and gc
-    /// enabled (gc_depth = 3) and make sure that for the exact same DAG the
-    /// linearizer will commit different blocks according to the rules.
-    #[rstest]
-    #[tokio::test]
-    async fn test_handle_commit_with_gc_simple(#[values(0, 3)] gc_depth: u32) {
-        telemetry_subscribers::init_for_testing();
-
-        let num_authorities = 4;
-        let (mut context, _keys) = Context::new_for_test(num_authorities);
-        context.protocol_config.set_gc_depth_for_testing(gc_depth);
-
-        if gc_depth == 0 {
-            context
-                .protocol_config
-                .set_consensus_linearize_subdag_v2_for_testing(false);
-        }
-
-        let context = Arc::new(context);
-        let dag_state = Arc::new(RwLock::new(DagState::new(
-            context.clone(),
-            Arc::new(MemStore::new()),
-        )));
-        let leader_schedule = Arc::new(LeaderSchedule::new(
-            context.clone(),
-            LeaderSwapTable::default(),
-        ));
-        let mut linearizer = Linearizer::new(context.clone(), dag_state.clone(), leader_schedule);
+    #[cfg(any())]
+    mod gc_tests {
+        /// This test will run the linearizer with GC disabled (gc_depth = 0)
+        /// and gc enabled (gc_depth = 3) and make sure that for the
+        /// exact same DAG the linearizer will commit different blocks
+        /// according to the rules.
+        #[rstest]
+        #[tokio::test]
+        async fn test_handle_commit_with_gc_simple(#[values(0, 3)] gc_depth: u32) {
```

Comment:
Remove the GC version of this test and leave the non-gc version as suggested for other files.